src/meili/map_matcher.cc

```diff
@@ -235,10 +235,10 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
   }
 
   // If we failed to get a valid edge or can't find it
-  // At least we know which point it matches
+  // At least we know which point it matches, to make leg builder
+  // work properly we snap the node at the end of the previous state
   const auto& edge = state.candidate().edges.front();
-  return {edge.projected,     std::sqrt(edge.distance), edgeid,
-          edge.percent_along, measurement.epoch_time(), stateid};
+  return {edge.projected, std::sqrt(edge.distance), edgeid, 1.0, measurement.epoch_time(), stateid};
```

Comment:
by changing the edge.percent_along (0.0) in old logic to 1.0. The match result returned will be put on the edge of the last edge of the previous state.

An alternative fix could be:
return {edge.projected, std::sqrt(edge.distance), edge.id, edge.percent_along, measurement.epoch_time(), stateid};
However, by doing so, we break the unit tests in trace attributes. Since this will generate empty path edges from meili mapmatching return. We currently do not handle such cases.